malpolon/data/datasets/poverty_dataset.py

```diff
@@ -0,0 +1,170 @@
+import pandas as pd
+from matplotlib import pyplot
+
+import torch
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+from torchvision import transforms
+
+from malpolon.data.data_module import BaseDataModule
+
+import datetime
+
+
+class PovertyDataModule(BaseDataModule):
+    def __init__(
+            self,
+            tif_dir: str = 'landsat_tiles/',
+            dataset_path: str = 'examples/poverty/dataset/',
+            labels_name: str = 'observations_2013+.csv',
+            train_batch_size: int = 32,
+            inference_batch_size: int = 16,
+            num_workers: int = 8,
+            **kwargs
+    ):
+
+        """DataModule for the Poverty dataset.
+        Args:
+            tif_dir (str): directory containing the tif files
+            dataset_path (str): path to the dataset
+            labels_name (str): name of the csv file containing the labels
+            train_batch_size (int): batch size for training
+            inference_batch_size (int): batch size for inference
+            num_workers (int): number of workers for the DataLoader
+            fold (int): fold to use for training
+            transform (torchvision.transforms): transform to apply to the data"""
+
+        super().__init__()
+        dataframe = pd.read_csv(dataset_path + labels_name)
+        self.dataframe = dataframe
+        self.dataframe_train = dataframe[dataframe['subset'] == 'train']
+        self.dataframe_val = dataframe[dataframe['subset'] == 'val']
+        self.dataframe_test = dataframe[dataframe['subset'] == 'test']
+        self.tif_dir = dataset_path + tif_dir
+        self.train_batch_size = train_batch_size
+        self.inference_batch_size = inference_batch_size
+        self.dict_normalize = json.load(open('examples/poverty/mean_std_normalize.json', 'r'))
+        self.num_workers = num_workers
+        self.task = 'regression'
+
+    def train_transform(self) -> Callable:
+        return transforms.Compose([
+            transforms.CenterCrop(224),
+            transforms.RandomHorizontalFlip(),
+            transforms.RandomVerticalFlip(),
+            transforms.Normalize(mean=self.dict_normalize['mean'], std=self.dict_normalize['std']),
+        ])
+
+    def test_transform(self) -> Callable:
+        return transforms.Compose([
+            transforms.CenterCrop(224),
+            transforms.Normalize(mean=self.dict_normalize['mean'], std=self.dict_normalize['std']),
+        ])
+
+    def get_dataset(self, split: str, **kwargs) -> Dataset:
+        if split == 'train':
+            dataset = MSDataset(self.dataframe_train, self.tif_dir, transform=self.train_transform())
+        elif split == 'val':
+            dataset = MSDataset(self.dataframe_val, self.tif_dir, transform=self.train_transform())
+        elif split == 'test':
+            dataset = MSDataset(self.dataframe_test, self.tif_dir, transform=self.test_transform())
+        return dataset
+
+    def train_dataloader(self):
```

Comment:
As your code is, it is only useful to re-define train/val/test_dataloaders to use `persistent_workers=True`, which is relevant if your epochs are fast and you want to make your training faster by avoiding to re-create workers at each epoch. This is done at the cost of memory.

If this is the intended use, ignore this comment. In the contrary case, you can remove altogether the re-defined train/val/test_dataloaders

Reply:
As we discussed, I removed the dataset from the PR.
I did make the chage though, thanks for the advice!